src/agent/agent.c

```diff
@@ -465,9 +465,10 @@ bool agent_parse_config(Agent *agent, const char *configfile) {
                         agent->config,
                         CFG_AGENT_DEFAULT_CONFIG,
                         CFG_ETC_HIRTE_AGENT_CONF,
-                        NULL); // TODO: https://github.com/containers/hirte/issues/148
-        if (result != 0 && result != -ENOENT) {
-                fprintf(stderr, "Error loading configuration: '%s'.\n", strerror(-result));
+                        CFG_ETC_AGENT_CONF_DIR);
+        if (result != 0) {
+                fprintf(stderr, "Error loading configuration: '%s'.", strerror(result));
```

Comment:
```suggestion
                fprintf(stderr, "Error loading configuration: '%s'.", strerror(-result));
```
We are returning error codes such as `-ENOENT`. Note the negation in front, which is used to identify errors. For `strerror` to find the correct description for it, we need to revert the negation again - `-result` in this case.

Reply:
Added -result